app/src/main/java/org/coepi/android/ui/onboarding/OnboardingFragment.kt

```diff
@@ -1,24 +1,109 @@
 package org.coepi.android.ui.onboarding
 
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
 import android.os.Bundle
-import android.text.method.LinkMovementMethod
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import kotlinx.android.synthetic.main.fragment_onboarding.*
+import org.coepi.android.R.string
 import org.coepi.android.databinding.FragmentOnboardingBinding.inflate
+import org.coepi.android.ui.onboarding.OnboardingCard.LargeOnboardingCard
+import org.coepi.android.ui.onboarding.OnboardingCard.SmallOnboardingCard
+import org.coepi.android.ui.onboarding.OnboardingClickEvent.FaqClicked
+import org.coepi.android.ui.onboarding.OnboardingClickEvent.JoinClicked
+import org.coepi.android.ui.onboarding.OnboardingClickEvent.NextClicked
+import org.coepi.android.ui.onboarding.OnboardingClickEvent.PrivacyLinkClicked
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class OnboardingFragment : Fragment() {
+
     private val viewModel by viewModel<OnboardingViewModel>()
 
+    private val compositeDisposable = CompositeDisposable()
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? = inflate(inflater, container, false).apply {
         lifecycleOwner = viewLifecycleOwner
         vm = viewModel
 
-        onboardingLink.movementMethod = LinkMovementMethod.getInstance()
     }.root
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+
+        onboarding_info_recycler.layoutManager =
+            object :LinearLayoutManager(context, HORIZONTAL, false) {
+                override fun canScrollHorizontally(): Boolean = false
+            }
+
+        onboarding_info_recycler.adapter =
+            OnboardingAdapter(
+                initializeOnboardingCards(requireContext())
+            ).apply {
+                onboardingEventStream
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe {
+                        when (it) {
+                            is NextClicked ->
+                                onboarding_info_recycler.layoutManager?.scrollToPosition(it.currentPosition + 1)
+
+                            is PrivacyLinkClicked -> startActivity(
+                                Intent(Intent.ACTION_VIEW, Uri.parse(PRIVACY_WEBPAGE_URL))
+                            )
+
+                            is JoinClicked -> viewModel.onCloseClick()
+
+                            is FaqClicked -> startActivity(
+                                Intent(Intent.ACTION_VIEW, Uri.parse(FAQ_WEBPAGE_URL)
+                                )
```

Comment:
Logic should be in the view model. Add separate observables / live data there for navigating to scroll position, opening activity, etc. Fragment should have only the minimum to wire with the UI.